File: Java/mergesort/gemma2-9b-it/Gemma2_9b_it.java
import java.util.Random;
public class Gemma2_9b_it {

    public static void mergeSort(int[] arr) {
        if (arr.length <= 1) {
            return; // Base case: already sorted
        }

        int mid = arr.length / 2;
        int[] left = new int[mid];
        int[] right = new int[arr.length - mid];

        // Copy data to subarrays
        for (int i = 0; i < mid; i++) {
            left[i] = arr[i];
        }
        for (int i = mid; i < arr.length; i++) {
            right[i - mid] = arr[i];
        }

        // Recursively sort subarrays
        mergeSort(left);
        mergeSort(right);

        // Merge sorted subarrays
        merge(arr, left, right);
    }

    private static void merge(int[] arr, int[] left, int[] right) {
        int i = 0, j = 0, k = 0;

        while (i < left.length && j < right.length) {
            if (left[i] <= right[j]) {
                arr[k++] = left[i++];
            } else {
                arr[k++] = right[j++];
            }
        }

        // Copy remaining elements from left subarray
        while (i < left.length) {
            arr[k++] = left[i++];
        }

        // Copy remaining elements from right subarray
        while (j < right.length) {
            arr[k++] = right[j++];
        }
    }

    public static void main(String[] args) {
        Random random = new Random();

        // Tamanho do vetor
        int size = 1000;

        // Inicializando o vetor com 1000 números aleatórios entre 1 e 1000
        int[] arr = new int[size];
        for (int i = 0; i < size; i++) {
            arr[i] = random.nextInt(1000) + 1; // Gera números entre 1 e 1000
        }

        // Realizando a ordenação com MergeSort
        mergeSort(arr);

    }
}

File: Python/mergesort/gemma2-9b-it/gemma2-9b-it.py
def merge_sort(arr):
    """
    Sorts a list using the Merge Sort algorithm.

    Args:
        arr: The list to be sorted.

    Returns:
        The sorted list.
    """

    if len(arr) <= 1:
        return arr

    mid = len(arr) // 2
    left_half = merge_sort(arr[:mid])
    right_half = merge_sort(arr[mid:])

    return merge(left_half, right_half)

def merge(left, right):
    """
    Merges two sorted lists into a single sorted list.

    Args:
        left: The first sorted list.
        right: The second sorted list.

    Returns:
        A new sorted list containing all elements from both input lists.
    """

    merged = []
    i = 0
    j = 0

    while i < len(left) and j < len(right):
        if left[i] <= right[j]:
            merged.append(left[i])
            i += 1
        else:
            merged.append(right[j])
            j += 1

    # Append any remaining elements from the left or right list
    merged.extend(left[i:])
    merged.extend(right[j:])

    return merged

import random

def main():
    arr = [random.randint(0, 10000) for _ in range(1000)]
    merge_sort(arr)

if __name__ == "__main__":
    main()
